### ISSUE TITLE
[Enhancement]: Block-based checkout spacing improvements

### ISSUE THREAD

**Issue by ralucaStan** [woothemes/woocommerce#46004]
This is building off Tarun's recently completed work in #44160 . The items below specify further changes and mobile specific improvements.

- [ ] Change space between sections from 36px to 40px 2JyXNimI4oGeJZhgQNZE9V-fi-2072%3A31996

![CleanShot 2024-02-21 at 11 10 08@2x](https://github.com/woocommerce/woocommerce/assets/99638528/266345de-c782-4002-89e1-c93f6ba2c2eb)

- [ ] For mobile: Add separator between each section, with a 36px top and bottom margin 2JyXNimI4oGeJZhgQNZE9V-fi-2056%3A14274

<p align="center">
<img src="https://github.com/woocommerce/woocommerce/assets/99638528/1360ee5e-edbc-40fb-a1d8-5352a1dd140a" alt="First Image Description" width="400" />
<img src="https://github.com/woocommerce/woocommerce/assets/99638528/4a2aa263-86ec-4b01-82b4-558f6ef9cf43" alt="First Image Description" width="400" />

</p>

**Comment by ralucaStan** [comment]
The team expressed some concerns about the border that should appear on mobile. It might require some changes at the inner cart level, so there is a high chance we might not be following that design if the solution is not straightforward: low effort, no impact on blocks's structure

**Comment by opr** [comment]
Nikki and I discussed this today and we are going to increase the padding to 48px rather than 40px.

We also noticed the following issues that need to be solved:
1. The spacing underneath the terms and conditions block should be increased in case that block is moved. Currently there is no space under it.
2. The separator between the order notes block and terms and conditions block should be removed and in its place an independent separator block should be added.
    1. This block should be styled like the existing one, in light grey.
    2. The default separator block should not be affected by this.
    3. The merchant should be able to move/remove the order notes, checkout terms and conditions, and separator.

Next steps for this issue are to address point 1 above in https://github.com/woocommerce/woocommerce/pull/47565 and for @nikkivias to review

Create a follow-up issue for point 2 as it would add complexity to the existing PR.

Follow up issue is created: https://github.com/woocommerce/woocommerce/issues/47910
